diff --git a/tools/objtool/check.c b/tools/objtool/check.c
index 21735829b860..8c296324e044 100644
--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -2833,7 +2833,14 @@ static inline const char *call_dest_name(struct instruction *insn)
 	rel = insn_reloc(NULL, insn);
 	if (rel && !strcmp(rel->sym->name, "pv_ops")) {
 		idx = (rel->addend / sizeof(void *));
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+#endif
 		snprintf(pvname, sizeof(pvname), "pv_ops[%d]", idx);
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
 		return pvname;
 	}
